lib/steep/interface/builder.rb

```diff
@@ -725,6 +727,40 @@ def proc_shape(proc, proc_shape)
         shape
       end
 
+      def replace_guard_method(definition, method_def, method_type)
+        match = method_def.annotations.filter_map { AST::Types::Logic::Guard::PATTERN.match(_1.string) }.first
+        if match
+          subject = match[1] or raise
+          operator = match[2] or raise
+          type_name = match[3] or raise
+
+          type = RBS::Parser.parse_type(type_name)
+          raise "Unknown type: #{type_name}" unless type
+
+          context = context_from(definition.type_name)
+          type = type.map_type_name { factory.absolute_type_name(_1, context: context) or raise "Unknown type: #{_1}" }
```

Comment:
Is there a better way to handle errors on Interface::Builder? It seems there is no `typing` object. So `typing.add_error()` can't be used here.

Reply:
Finally, I implemented these checks in signature/validator.rb.